diff --git a/node_modules/sucrase/dist/esm/parser/traverser/statement.js b/node_modules/sucrase/dist/esm/parser/traverser/statement.js
index 34a6511..4f19dcc 100644
--- a/node_modules/sucrase/dist/esm/parser/traverser/statement.js
+++ b/node_modules/sucrase/dist/esm/parser/traverser/statement.js
@@ -89,7 +89,7 @@ import {
 } from "./util";
 
 export function parseTopLevel() {
-  parseBlockBody(tt.eof);
+  parseBlockBody(tt.eof, true);
   state.scopes.push(new Scope(0, state.tokens.length, true));
   if (state.scopeDepth !== 0) {
     throw new Error(`Invalid scope depth at end of file: ${state.scopeDepth}`);
@@ -104,7 +104,7 @@ export function parseTopLevel() {
 // `if (foo) /blah/.exec(foo)`, where looking at the previous token
 // does not help.
 
-export function parseStatement(declaration) {
+export function parseStatement(declaration, isTopLevel) {
   if (isFlowEnabled) {
     if (flowTryParseStatement()) {
       return;
@@ -113,10 +113,10 @@ export function parseStatement(declaration) {
   if (match(tt.at)) {
     parseDecorators();
   }
-  parseStatementContent(declaration);
+  parseStatementContent(declaration, isTopLevel);
 }
 
-function parseStatementContent(declaration) {
+function parseStatementContent(declaration, isTopLevel) {
   if (isTypeScriptEnabled) {
     if (tsTryParseStatementContent()) {
       return;
@@ -124,6 +124,7 @@ function parseStatementContent(declaration) {
   }
 
   const starttype = state.type;
+  const startTokenIndex = state.tokens.length
 
   // Most types of statements are recognized by the keyword they
   // start with. Many are trivial to parse, some require a bit of
@@ -147,11 +148,13 @@ function parseStatementContent(declaration) {
       if (lookaheadType() === tt.dot) break;
       if (!declaration) unexpected();
       parseFunctionStatement();
+      state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
 
     case tt._class:
       if (!declaration) unexpected();
       parseClass(true);
+      state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
 
     case tt._if:
@@ -159,9 +162,11 @@ function parseStatementContent(declaration) {
       return;
     case tt._return:
       parseReturnStatement();
+      state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
     case tt._switch:
       parseSwitchStatement();
+      state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
     case tt._throw:
       parseThrowStatement();
@@ -247,11 +252,15 @@ function parseStatementContent(declaration) {
       simpleName = token.contextualKeyword;
     }
   }
+  
+  state.tokens[startTokenIndex].isTopLevel = isTopLevel;
+
   if (simpleName == null) {
     semicolon();
     return;
   }
   if (eat(tt.colon)) {
+    state.tokens[startTokenIndex].isTopLevel = isTopLevel;
     parseLabeledStatement();
   } else {
     // This was an identifier, so we might want to handle flow/typescript-specific cases.
@@ -581,9 +590,9 @@ export function parseBlock(isFunctionScope = false, contextId = 0) {
   state.scopeDepth--;
 }
 
-export function parseBlockBody(end) {
+export function parseBlockBody(end, isTopLevel) {
   while (!eat(end) && !state.error) {
-    parseStatement(true);
+    parseStatement(true, isTopLevel);
   }
 }
 
diff --git a/node_modules/sucrase/dist/parser/traverser/statement.js b/node_modules/sucrase/dist/parser/traverser/statement.js
index 6be3391..255fbb5 100644
--- a/node_modules/sucrase/dist/parser/traverser/statement.js
+++ b/node_modules/sucrase/dist/parser/traverser/statement.js
@@ -89,7 +89,7 @@ var _lval = require('./lval');
 var _util = require('./util');
 
  function parseTopLevel() {
-  parseBlockBody(_types.TokenType.eof);
+  parseBlockBody(_types.TokenType.eof, true);
   _base.state.scopes.push(new (0, _state.Scope)(0, _base.state.tokens.length, true));
   if (_base.state.scopeDepth !== 0) {
     throw new Error(`Invalid scope depth at end of file: ${_base.state.scopeDepth}`);
@@ -104,7 +104,7 @@ var _util = require('./util');
 // `if (foo) /blah/.exec(foo)`, where looking at the previous token
 // does not help.
 
- function parseStatement(declaration) {
+ function parseStatement(declaration, isTopLevel) {
   if (_base.isFlowEnabled) {
     if (_flow.flowTryParseStatement.call(void 0, )) {
       return;
@@ -113,10 +113,10 @@ var _util = require('./util');
   if (_tokenizer.match.call(void 0, _types.TokenType.at)) {
     parseDecorators();
   }
-  parseStatementContent(declaration);
+  parseStatementContent(declaration, isTopLevel);
 } exports.parseStatement = parseStatement;
 
-function parseStatementContent(declaration) {
+function parseStatementContent(declaration, isTopLevel) {
   if (_base.isTypeScriptEnabled) {
     if (_typescript.tsTryParseStatementContent.call(void 0, )) {
       return;
@@ -124,6 +124,7 @@ function parseStatementContent(declaration) {
   }
 
   const starttype = _base.state.type;
+  const startTokenIndex = _base.state.tokens.length
 
   // Most types of statements are recognized by the keyword they
   // start with. Many are trivial to parse, some require a bit of
@@ -147,11 +148,13 @@ function parseStatementContent(declaration) {
       if (_tokenizer.lookaheadType.call(void 0, ) === _types.TokenType.dot) break;
       if (!declaration) _util.unexpected.call(void 0, );
       parseFunctionStatement();
+      _base.state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
 
     case _types.TokenType._class:
       if (!declaration) _util.unexpected.call(void 0, );
       parseClass(true);
+      _base.state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
 
     case _types.TokenType._if:
@@ -159,6 +162,7 @@ function parseStatementContent(declaration) {
       return;
     case _types.TokenType._return:
       parseReturnStatement();
+      _base.state.tokens[startTokenIndex].isTopLevel = isTopLevel;
       return;
     case _types.TokenType._switch:
       parseSwitchStatement();
@@ -210,6 +214,7 @@ function parseStatementContent(declaration) {
         if (_tokenizer.match.call(void 0, _types.TokenType._function) && !_util.canInsertSemicolon.call(void 0, )) {
           _util.expect.call(void 0, _types.TokenType._function);
           parseFunction(functionStart, true);
+          _base.state.tokens[startTokenIndex].isTopLevel = isTopLevel;
           return;
         } else {
           _base.state.restoreFromSnapshot(snapshot);
@@ -247,11 +252,15 @@ function parseStatementContent(declaration) {
       simpleName = token.contextualKeyword;
     }
   }
+
+  _base.state.tokens[startTokenIndex].isTopLevel = isTopLevel;
+
   if (simpleName == null) {
     _util.semicolon.call(void 0, );
     return;
   }
   if (_tokenizer.eat.call(void 0, _types.TokenType.colon)) {
+    _base.state.tokens[startTokenIndex].isTopLevel = false;
     parseLabeledStatement();
   } else {
     // This was an identifier, so we might want to handle flow/typescript-specific cases.
@@ -581,9 +590,9 @@ function parseIdentifierStatement(contextualKeyword) {
   _base.state.scopeDepth--;
 } exports.parseBlock = parseBlock;
 
- function parseBlockBody(end) {
+ function parseBlockBody(end, isTopLevel) {
   while (!_tokenizer.eat.call(void 0, end) && !_base.state.error) {
-    parseStatement(true);
+    parseStatement(true, isTopLevel);
   }
 } exports.parseBlockBody = parseBlockBody;
 
diff --git a/node_modules/sucrase/dist/types/parser/tokenizer/index.d.ts b/node_modules/sucrase/dist/types/parser/tokenizer/index.d.ts
index 45cd799..66b2480 100644
--- a/node_modules/sucrase/dist/types/parser/tokenizer/index.d.ts
+++ b/node_modules/sucrase/dist/types/parser/tokenizer/index.d.ts
@@ -51,6 +51,7 @@ export declare class Token {
     isOptionalChainEnd: boolean;
     subscriptStartIndex: number | null;
     nullishStartIndex: number | null;
+    isTopLevel?: boolean;
 }
 export declare function next(): void;
 export declare function nextTemplateToken(): void;
